internal/services/containers/container_registry_credential_set_resource.go

```diff
@@ -0,0 +1,304 @@
+			id, err := credentialsets.ParseCredentialSetID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			registryId := registries.NewRegistryID(id.SubscriptionId, id.ResourceGroupName, id.RegistryName)
+
+			var config ContainerRegistryCredentialSetModel
+			if err := metadata.Decode(&config); err != nil {
+				return err
+			}
+
+			config.Name = id.CredentialSetName
+			config.ContainerRegistryId = registryId.ID()
+
+			if model := resp.Model; model != nil {
+				config.Identity = flattenIdentity(model.Identity)
+				if props := model.Properties; props != nil {
+					config.LoginServer = pointer.From(props.LoginServer)
+					config.AuthenticationCredential = flattenAuthCredentials(props.AuthCredentials)
+				}
+			}
+			return metadata.Encode(&config)
+		},
+	}
+}
+
+func (ContainerRegistryCredentialSetResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Containers.CredentialSetsClient
+			id, err := credentialsets.ParseCredentialSetID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (ContainerRegistryCredentialSetResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return credentialsets.ValidateCredentialSetID
+}
+
+func expandAuthCredentials(input []AuthenticationCredential) *[]credentialsets.AuthCredential {
+	output := make([]credentialsets.AuthCredential, 0)
+	if len(input) == 0 {
+		return &output
+	}
+	for _, v := range input {
+		output = append(output, credentialsets.AuthCredential{
+			Name:                     pointer.To(credentialsets.CredentialNameCredentialOne),
+			UsernameSecretIdentifier: pointer.To(v.UsernameSecretId),
+			PasswordSecretIdentifier: pointer.To(v.PasswordSecretId),
+		})
+	}
+	return &output
+}
+
+func flattenAuthCredentials(input *[]credentialsets.AuthCredential) []AuthenticationCredential {
+	output := make([]AuthenticationCredential, 0)
+	if input == nil {
+		return output
+	}
+	for _, v := range *input {
+		output = append(output, AuthenticationCredential{
+			UsernameSecretId: pointer.From(v.UsernameSecretIdentifier),
+			PasswordSecretId: pointer.From(v.PasswordSecretIdentifier),
+		})
+	}
+	return output
+}
+
+// read the note [1] above why we transform the identity here like that
+func flattenIdentity(input *identity.SystemAndUserAssignedMap) []identity.ModelSystemAssigned {
+	if input == nil {
+		return nil
+	}
+	output := make([]identity.ModelSystemAssigned, 1)
+	output[0] = identity.ModelSystemAssigned{
+		Type:        input.Type,
```

Comment:
We call a normalising function on the identity type returned here in the flatten functions of the `identity` package because the API is often inconsistent with the casing

https://github.com/hashicorp/go-azure-helpers/blob/624ef4f92d9f94a4a467b2bf5fb79974ef6c3255/resourcemanager/identity/constants.go#L20-L38

We can do this and it should take care of the casing issue for us 
```suggestion
		Type:        identity.Type(input.Type),
```

Reply:
I tested that and it did not resolve the issue 🫤